File: 299. Bulls and Cows/299_test.go
package leetcode

import(
	"testing"
)

func TestGetHint(t *testing.T){
	type test struct{
		input1 string
		input2 string
		want string
	}
	
	tests := map[string]test{
	"1": test{"1807","7810","1A3B"},
	"2": test{"1123","0111","1A1B"},
	}
	
	for name, tc := range tests{
		t.Run(name, func(t *testing.T){
			got := getHint(tc.input1, tc.input2)
			if got != tc.want{
				t.Errorf("Test Failed! want: %#v got: %#v", tc.want, got)
			}
		})
	}
}

func BenchmarkSingleNumber(b *testing.B){
	for i:=0; i<b.N; i++{
		getHint("1807","7810")
	}
}

File: 299. Bulls and Cows/_299_test.py
import unittest
from _299 import *

class test():
    def __init__(self, input1, input2, want):
        self.input1 = input1
        self.input2 = input2
        self.want = want

class myTest(unittest.TestCase):
    def testGetHint(self):
        so = Solution()

        test1 = test("1807", "7810", "1A3B")
        test2 = test("1123", "0111", "1A1B")

        got1 = so.getHint(test1.input1, test1.input2)
        got2 = so.getHint(test2.input1, test2.input2)

        self.assertEqual(got1, test1.want, "Test1 Failed!")
        self.assertEqual(got2, test2.want, "Test1 Failed!")

if __name__ == "__main__":
    testSuite = unittest.TestSuite()
    testSuite.addTest(myTest("testGetHint"))
    unittest.TextTestRunner(testSuite)